workflow/automation/lib/MgmtDB.py

```diff
@@ -351,14 +355,14 @@ def add_retries(self, n_max_retries: int):
                 failure_count.update({key: {"killed_WCT": 0, "failed": 0}})
             failure_count[key][state] += 1
 
-        with connect_db_ctx(self._db_file) as cur:
-            for key, fail_count in failure_count.items():
-                if any([x >= n_max_retries for x in fail_count.values()]):
-                    continue
-                run_name, proc_type = key.split("__")
-                # Gets the number of entries for the task with state in [created, queued, running or completed]
-                # Where completed has enum index 5, and the other 4 less than this
-                # If any are found then don't add another entry
+        for key, fail_count in failure_count.items():
+            if any([x >= n_max_retries for x in fail_count.values()]):
+                continue
+            run_name, proc_type = key.split("__")
+            # Gets the number of entries for the task with state in [created, queued, running or completed]
+            # Where completed has enum index 5, and the other 4 less than this
+            # If any are found then don't add another entry
```

Comment:
I find explanation confusing. What does enum mean? Why must the other four (three? created, queued, running?) have index less than 5?

Reply:
https://github.com/ucgmsim/qcore/blob/f57aae21c1489315369c9791a851c991ffb06c4c/qcore/constants.py#L448
![Screenshot 2024-05-31 at 2 45 49 PM](https://github.com/ucgmsim/slurm_gm_workflow/assets/466989/7067c569-bebc-4b73-a381-20b0d8baffeb)